### PR TITLE
Minor Electrical Optimisations

### PR DESCRIPTION
**Pull request by ThatDan123**
-Removed unnecessary dictionary

-Removed resistance property (properties do have a performance difference from just using a public variable)

Would be nice in the future to rewrite the whole system to be burst compatible

### CODE REVIEW

**Review by NoooneyDude — DISMISSED** [review]

**Comment by NoooneyDude** [comment]
> -Removed resistance property (properties do have a performance difference from just using a public variable)

Don't suppose you could share a link where I can learn more about this? Just curious and little surprised, surely the compiler w/ optimisations enabled would handle it.

Edit: just read #general-dev. Right, we're running dev mode on the server. All good 👍

File: UnityProject/Assets/Scripts/Objects/Engineering/APCPoweredDevice.cs [hunk 1]
```diff
@@ -60,14 +62,8 @@ public class APCPoweredDevice : NetworkBehaviour, IServerDespawn, IEmpAble, IMul
 		[FormerlySerializedAs("resistance")]
 		private float InitialResistance = 99999999;
 
-
-		private float resistance = 99999999;
-
-
-		public float Resistance {
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Objects/Engineering/APCPoweredDevice.cs:67]: I think this was meant to trigger resistance change?

File: UnityProject/Assets/Scripts/Systems/Electricity/FunctionsAndClasses/ElectricityFunctions.cs [hunk 2]
```diff
@@ -185,76 +185,49 @@ public static float WorkOutResistance(Dictionary<IntrinsicElectronicData, VIRRes
...
-							AnInterestingDictionary[CurrentItem.Key] = (float) -CurrentItem.Value.Current();
-						}
-					}
+					current += (float) currentItem.Value.Current();
 				}
 
-				foreach (var CurrentItem in AnInterestingDictionary)
+				foreach (var currentItem in currentIDItem.Value.CurrentGoingTo)
 				{
-					if (CurrentItem.Value > 0)
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Systems/Electricity/FunctionsAndClasses/ElectricityFunctions.cs:234]: was there any scenario where this was false?  might be worth putting in else and then putting a breakpoint to see if anything triggers it

Comment (ThatDan123) [reply UnityProject/Assets/Scripts/Systems/Electricity/FunctionsAndClasses/ElectricityFunctions.cs:234]: Added a check for it

File: UnityProject/Assets/Scripts/Objects/Engineering/APCPoweredDevice.cs (hunk 1 above)
Comment (ThatDan123) [reply UnityProject/Assets/Scripts/Objects/Engineering/APCPoweredDevice.cs:67]: Can't find any function in the class for a resistance change?

**Review by Bod9001 — APPROVED** [review]

**Pull request merged**